File: trunk/PO210641/task_03/src/Item.cpp
#include "item.h"

Item::Item (int id, int durability, int age, double cost, double weight)
	: id(id), durability(durability), age(age), cost(cost), weight(weight) {
}

double Item::CalculateCostPerGram() const
{
    return cost / weight;
}

int Item::CalculateRemainingLifetime() const
{
    return durability - age;
}

void Item::Show() const
{
    std::cout << "Id: " << id << "\n";
    std::cout << "Cost: $" << cost << "\n";
    std::cout << "Weight: " << weight << " grams\n";
    std::cout << "Durability: " << durability << " years\n";
    std::cout << "Age: " << age << " years\n";
}

File: trunk/PO210649/task_03/ConsoleApp2/ClassLibrary1/Item.cs
namespace ClassLibrary1;

public class Item
{
    private int _id;
    private decimal _cost;
    private int _weight;
    private int _durability;
    private int _age;

    public Item(int id, decimal cost, int weight, int durability, int age)
    {
        _id = id;
        _cost = cost;
        _weight = weight;
        _durability = durability;
        _age = age;
    }

    public decimal CostForWeight()
    {
        return _weight / _cost;
    }

    public int TimeOfLife()
    {
        return _durability - _age;
    }

    public void ShowInfo()
    {
        Console.WriteLine($"Id: {_id}\n" +
                          $"Cost: {_cost}\n" +
                          $"Weight: {_weight}\n" +
                          $"Durability: {_durability}\n" +
                          $"Age: {_age}");
    }
}